test/evergreen.yml

```diff
@@ -296,12 +299,15 @@ functions:
           powershell.exe  -NonInteractive '.\test\evergreen\build_windows.ps1' -configure 1 $DEFINED_EVERGREEN_CONFIG_FLAGS ${windows_configure_flags|}
         else
           echo "Using config flags $DEFINED_EVERGREEN_CONFIG_FLAGS ${posix_configure_flags|}"
-          # Fetch the gperftools library if needed. This will also get tcmalloc.
-          if [[ ${ENABLE_TCMALLOC|0} -eq 1 ]] && [ ! -d "./automation-scripts" ]; then
-            is_cmake_build=true
-            git clone git@github.com:wiredtiger/automation-scripts.git
-            . automation-scripts/evergreen/find_gperftools.sh ${s3_access_key} ${s3_secret_key} ${build_variant} $is_cmake_build
+
+          if [[ ${ENABLE_TCMALLOC|0} -eq 1 ]]; then
+            bin/bash test/evergreen/tcmalloc_install_or_build.sh ${build_variant}
+
+            # Preload tcmalloc in the spawnhost environment: this will affect
```

Comment:
"spawnhost environment" reads a bit weird. Does it mean the "Evergreen environment/host" in general? Spawnhost is usually referred to in the context of failure troubleshooting (or for daily development purposes).